### PR TITLE
fails kfctl if overlays not found

### PR DESCRIPTION
**Pull request by adrian555**
Toward #155. `kfctl` will fail and return 500 if component's overlay dir is not found.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfctl/186)

### CODE REVIEW

File: third_party/license_info.csv [hunk 1]
```diff
@@ -317,11 +317,11 @@ kubernetes-sigs/yaml,https://github.com/kubernetes-sigs/yaml/blob/master/LICENSE
...
-modernc.org/golex,https://gitlab.com/cznic/golex/blob/master/LICENSE,BSD 3-Clause "New" or "Revised" License,https://gitlab.com/cznic/golex/blob/master/LICENSE
-modernc.org/mathutil,https://gitlab.com/cznic/mathutil/blob/master/LICENSE,BSD 3-Clause "New" or "Revised" License,https://gitlab.com/cznic/mathutl/blob/master/LICENSE
-modernc.org/strutil,https://gitlab.com/cznic/strutil/blob/master/LICENSE,BSD 3-Clause "New" or "Revised" License,https://gitlab.com/cznic/strutil/blob/master/LICENSE
-modernc.org/xc,https://gitlab.com/cznic/xc/blob/master/LICENSE,BSD 3-Clause "New" or "Revised" License,https://gitlab.com/cznic/xc/blob/master/LICENSE
-heketi/utils,https://github.com/heketi/heketi/blob/master/LICENSE,Apache License 2.0,https://github.com/heketi/heketi/blob/master/LICENSE
-kr/logfmt,https://github.com/kr/logfmt/blob/master/Readme,MIT License,https://github.com/kr/logfmt/blob/master/Readme
+bitbucket.org/bertimus9/systemstat,https://bitbucket.org/bertimus9/systemstat/raw/master/LICENSE,MIT License,https://bitbucket.org/bertimus9/systemstat/raw/master/LICENSE
```
Comment (animeshsingh) [inline third_party/license_info.csv]: This fix is already in this PR. Can we close PR #185 ?

File: pkg/kfapp/kustomize/kustomize.go [hunk 2]
```diff
@@ -859,7 +859,10 @@ func MergeKustomizations(kfDef *kfconfig.KfConfig, compDir string, overlayParams
 				}
 			}
 		} else {
-			log.Warnf("No overlay %v for component at %v, skipping...", overlayParam, compDir)
+			return nil, &kfapisv3.KfError{
+				Code:    int(kfapisv3.INTERNAL_ERROR),
+				Message: fmt.Sprintf("no overlay %v for component at %v Error %v", overlayParam, compDir, err),
```
Comment (animeshsingh) [inline pkg/kfapp/kustomize/kustomize.go]: Maybe capitalize...? 
```suggestion
				Message: fmt.Sprintf("No overlay %v for Component at %v Error %v", overlayParam, compDir, err),
```

Comment (adrian555) [reply pkg/kfapp/kustomize/kustomize.go]: Not sure about whether there is a convention or not. But I was following the one fmt.Sprintf right above (line 858).

File: third_party/license_info.csv (hunk 1 above)
Comment (adrian555) [reply third_party/license_info.csv]: sure

File: pkg/kfapp/kustomize/kustomize.go (hunk 2 above)
Comment (animeshsingh) [reply pkg/kfapp/kustomize/kustomize.go]: Ok - even there "N" is capital in "no"

Comment (adrian555) [reply pkg/kfapp/kustomize/kustomize.go]: What I mentioned is this one:
https://github.com/kubeflow/kfctl/blob/master/pkg/kfapp/kustomize/kustomize.go#L858

File: third_party/license_info.csv (hunk 1 above)
Comment (animeshsingh) [reply third_party/license_info.csv]: vis a vis @jlewi  comment on previous PR, these third party licenses are regenerated I believe, and hence the change is needed?

File: pkg/kfapp/kustomize/kustomize.go (hunk 2 above)
Comment (animeshsingh) [reply pkg/kfapp/kustomize/kustomize.go]: hmmm - @jlewi is there a convention here?

File: third_party/license_info.csv (hunk 1 above)
Comment (adrian555) [reply third_party/license_info.csv]: This `license_info.csv` file has the url links to download the license texts. The current one contains wrong download links and hence the `make test` fails at license check. Getting this fixed the earlier the better I believe.

**Comment by animeshsingh** [comment]
@adrian555 remove the license files from here and bring in a different PR - and we can move this forward.

**Comment by adrian555** [comment]
@animeshsingh license related change removed.

**Comment by animeshsingh** [comment]
thanks @adrian555 
Once the test passes, this can be good to go

/lgtm
/approve

**Pull request merged**